projects/js-packages/publicize-components/src/utils/share-status.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Normalizes the share status object.
+ * TODO: Optimize this function to avoid unnecessary iterations.
+ *
+ * @param {import( './types' ).PostShareStatus} shareStatus - Share status object.
+ * @return {import( './types' ).PostShareStatus | undefined} - Normalized share status object.
+ */
+export function normalizeShareStatus( shareStatus ) {
+	if ( ! shareStatus || ! ( 'shares' in shareStatus ) || ! shareStatus.done ) {
+		return;
+	}
+	// Sort shares to show the latest shares on the top.
+	shareStatus.shares.sort( ( a, b ) => b.timestamp - a.timestamp );
+
+	// Remove failed shares that have a successful share later.
+	shareStatus.shares = shareStatus.shares.filter( share => {
+		const hasSuccessfulShareLater = shareStatus.shares.some( otherShare => {
+			return (
+				otherShare.timestamp > share.timestamp &&
+				'success' === otherShare.status &&
+				otherShare.external_id === share.external_id &&
+				share.external_id // We added external_id later to the object
+			);
+		} );
+
+		if ( 'failure' === share.status ) {
+			return ! hasSuccessfulShareLater;
+		}
+
+		return true;
+	} );
```

Comment:
If we want to always show everything this part needs to be changed, or removed.

If we want to hide Retry for those that was already retried, we can update the status here to be `retried` or something like that.

Reply:
Yeah, let's show the retry button as is and not hide them.

Reply:
OK, let us then leave it as is for now and may be come back later to improve if needed.

I think it does make sense to treat it as activity log.